compiler/rustc_infer/src/infer/type_variable.rs

```diff
@@ -284,6 +284,11 @@ impl<'tcx> TypeVariableTable<'_, 'tcx> {
         self.sub_root_var(a) == self.sub_root_var(b)
     }
 
+    /// Returns `true` if `a` and `b` have same "root" (i.e., `a == b`).
+    pub fn check_equal(&mut self, a: ty::TyVid, b: ty::TyVid) -> bool {
```

Comment:
This doesn't need to be a separate function

Reply:
Removed